pkg/log/log.go

```diff
@@ -188,7 +214,18 @@ func NewTestLoggerPtr() *Logger {
 // Msgf logs the event with a formatted message.
 func (e *Event) Msgf(format string, args ...interface{}) {
 	msg := fmt.Sprintf(format, args...)
-	e.logger.LogAttrs(nil, e.level, msg, e.attrs...)
+	// Add caller and function info to attributes if captured
+	attrs := e.attrs
+
+	if e.caller != "" {
+		attrs = append(attrs, slog.String("caller", e.caller))
+	}
+
+	if e.function != "" {
+		attrs = append(attrs, slog.String("func", e.function))
+	}
+
+	e.logger.LogAttrs(nil, e.level, msg, attrs...)
```

Comment:
Do these changes also cover point 1 I was asking about here? https://github.com/project-zot/zot/pull/3405#issuecomment-3344341098
It would be nice if they do, but not a blocker or anything.